services/repository-managers/src/main/java/org/sagebionetworks/repo/util/jrjc/JiraClientImpl.java

```diff
@@ -3,73 +3,193 @@
  */
 public class JiraClientImpl implements JiraClient {
 	private static final String JIRA_URL = "https://sagebionetworks.jira.com";
 
-	private JiraRestClient restClient;
-	
-	private static <T> T waitForPromise(Promise<T> promise) {
-		try {
-			while (!promise.isDone()) {
-				Thread.sleep(100L);
-			}
-			return promise.get();
-		} catch (InterruptedException e) {
-			throw new RuntimeException(e);
-		} catch (ExecutionException e) {
-			throw new RuntimeException(e);
-		}
-	}
-	
+	private final SimpleHttpClient httpClient;
+	private String USERNAME;
+	private String APIKEY;
+	private static final Integer TIME_OUT = 30 * 1000; // 30 seconds
+	private static final String JIRA_API_PROJECT_URL = "/rest/api/3/project/";
+	private static final String JIRA_API_FIELDS_URL = "/rest/api/3/field/";
+	private static final String JIRA_API_ISSUE_URL = "/rest/api/3/issue/";
+	private static final String USER_AGENT = "Synapse";
+	private static JSONParser parser;
+	private static final String JIRA_PROJECT_ISSUE_TYPES_KEY = "issueTypes";
+	private static final String JIRA_PROJECT_ID_KEY = "id";
+
 	public JiraClientImpl() {
-    	final AsynchronousJiraRestClientFactory factory = new AsynchronousJiraRestClientFactory();
-    	URI jiraServerUri = URI.create(JIRA_URL);
-    	this.restClient = factory.createWithBasicHttpAuthentication(
-    			jiraServerUri, 
-    			StackConfigurationSingleton.singleton().getJiraUserEmail(), 
-    			StackConfigurationSingleton.singleton().getJiraUserApikey());
+		SimpleHttpClientConfig httpClientConfig = new SimpleHttpClientConfig();
+		httpClientConfig.setSocketTimeoutMs(TIME_OUT);
+		httpClient = new SimpleHttpClientImpl(httpClientConfig);
+
+		USERNAME = StackConfigurationSingleton.singleton().getJiraUserEmail();
+		APIKEY = StackConfigurationSingleton.singleton().getJiraUserApikey();
+		parser = new JSONParser();
 	}
 
 	@Override
-	public Project getProject(String projectKey) {
-		ProjectRestClient prc = restClient.getProjectClient();
-		return waitForPromise(prc.getProject(projectKey));
+	public JSONObject getProjectInfo(String projectKey, String issueTypeName) {
+		SimpleHttpRequest req = createRequest(JIRA_API_PROJECT_URL, "SG", "application/json");
+		SimpleHttpResponse resp = null;
+		try {
+			resp = httpClient.get(req);
+		} catch (IOException e) {
+			throw new RuntimeException();
+		}
+		handleResponseStatus(resp.getStatusCode());
+		String json = null;
+		json = resp.getContent();
+
+		Long issueTypeId = -1L;
+		String projectId = null;
+		try (Reader reader = new StringReader(json)) {
+			JSONObject pInfo = (JSONObject) parser.parse(reader);
+			JSONArray issueTypeNames = (JSONArray)pInfo.get(JIRA_PROJECT_ISSUE_TYPES_KEY);
+			Iterator<JSONObject> it = issueTypeNames.iterator();
+			while (it.hasNext()) {
+				JSONObject issueType = it.next();
+				String name = (String) issueType.get("name");
+				if (issueTypeName.equalsIgnoreCase(name)) {
+					issueTypeId = Long.parseLong((String) issueType.get("id"));
```

Comment:
use valueOf(), check that get() does not return null. And extract "id" as constant.